test/videojs.ads.test.js

```diff
@@ -1,1025 +1,732 @@
-  player.ads.startLinearAdMode();
-  equal(player.ads.state, 'ad-playback');
-  player.trigger('adserror');
-  equal(adend, 2, 'adend should have fired');
-});
-
-test('skipLinearAdMode in ad-playback does not trigger adskip', function(){
-  var adskip = 0;
-  player.on('adskip', function() {
-    adskip++;
-  });
-
-  equal(player.ads.state, 'content-set');
-  player.trigger('adsready');
-  equal(player.ads.state, 'ads-ready');
-  player.trigger('play');
-  player.ads.startLinearAdMode();
-  equal(player.ads.state, 'ad-playback');
-  player.ads.skipLinearAdMode();
-  equal(player.ads.state, 'ad-playback');
-  equal(adskip, 0, 'adskip event should not trigger when skipLinearAdMode called in ad-playback state');
-
-  player.ads.endLinearAdMode();
-  equal(player.ads.state, 'content-resuming');
-  equal(player.ads.triggerevent, 'adend', 'The reason for content-resuming should have been adend');
-
-  player.trigger('playing');
-  equal(player.ads.state, 'content-playback');
-  equal(contentPlaybackFired, 1, 'A content-playback event should have triggered');
-  equal(contentPlaybackReason, 'playing', 'The reason for content-playback should have been playing');
-});
-
-test('adsready in content-playback triggers readyforpreroll', function(){
-  expect(6);
-
-  player.on('readyforpreroll', function(event) {
-    equal(event.type, 'readyforpreroll', 'readyforpreroll should have been triggered.');
-  });
-
-  equal(player.ads.state, 'content-set');
-  player.trigger('play');
-  equal(player.ads.state, 'ads-ready?');
-  player.trigger('adtimeout');
-  equal(player.ads.state, 'content-playback');
-  equal(contentPlaybackFired, 1, 'A content-playback event should have triggered');
-  equal(contentPlaybackReason, 'adtimeout', 'The reason for content-playback should have been adtimeout');
-  player.trigger('adsready');
+  this.player.ads.contentSrc = 'http://example.com/movie-high.mp4';
+  this.player.src('http://example.com/movie-high.mp4');
+  this.player.trigger('loadstart');
+  assert.strictEqual(spy.callCount, 0, 'no contentupdate was generated');
+  assert.strictEqual(this.player.ads.state, 'content-playback', 'did not reset ad state');
+  assert.strictEqual(this.player.ads.contentSrc, this.player.currentSrc(), 'captures the current content src');
+});
+
+QUnit.test('the cancel-play timeout is cleared when exiting "preroll?"', function(assert) {
+  assert.expect(3);
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
+  assert.strictEqual(this.player.ads.state, 'preroll?', 'the player is waiting for prerolls');
+  this.player.tech.trigger('play');
+  assert.ok(this.clock.timers.hasOwnProperty(String(this.player.ads.cancelPlayTimeout)), 'a single cancel-play is registered');
+  this.player.tech.trigger('play');
+  this.player.tech.trigger('play');
+  this.player.tech.trigger('play');
+  assert.ok(this.clock.timers.hasOwnProperty(String(this.player.ads.cancelPlayTimeout)), 'a single cancel-play is registered');
+});
```

Comment:
It seems like this is doing an existence check, not verifying that a callback wasn't registered multiple times. Am I missing something?